backend/src/server/routes/v1/dashboard-router.ts

```diff
@@ -840,4 +840,91 @@ export const registerDashboardRouter = async (server: FastifyZodProvider) => {
       };
     }
   });
+
+  server.route({
+    method: "GET",
+    url: "/secrets-by-keys",
+    config: {
+      rateLimit: secretsLimit
+    },
+    schema: {
+      security: [
+        {
+          bearerAuth: []
+        }
+      ],
+      querystring: z.object({
+        projectId: z.string().trim(),
+        environment: z.string().trim(),
+        secretPath: z.string().trim().default("/").transform(removeTrailingSlash),
+        keys: z.string().trim().transform(decodeURIComponent)
```

Comment:
Is the `decodeURIComponent` transformer really necessary? You'd think this is something Axios or Fastify would handle automatically

Reply:
Just double checked but yeah; I think if they did it by default they could unintentionally mutate data that wasn't encoded but had `%20` for example